.github/workflows/build-and-test.yaml

```diff
@@ -2,8 +2,11 @@ name: Build-and-test
 
 on:
   push:
+    branches:
```

Comment:
Consider adding a filter for tags in addition to branches if the project uses version tags. This will ensure that CI processes are also run for tagged commits, which are often used for releases. [medium]